.githooks/commit-msg

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/sh
+
+declare -A components
+components=( \
+	["cnf-tests"]="cnf-tests" \
+	["features"]="feature-configs" \
+	["hack"]="hack" \
+	["hooks"]=".githooks" \
+	["ci"]="openshift-ci" \
+	["tools"]="tools" \
+	["s2i"]="tools/s2i-dpdk" \
+	["vendor"]="vendor" \
+	["ztp"]="ztp" 
+)
+exceptions=( go.mod go.sum LICENSE Makefile OWNERS OWNERS_ALIASES README.md )
+
+if ! [[ "$(head -1 $1)t sh" == *":"* ]]; then
+	echo ERROR: commit msg subject must include component name
+	exit 4
+fi
+msg_prefix=$(head -1 $1 | cut -d':' -f 1)
+for filename in $(git diff --name-only --cached); do
+	if [[ ! " ${exceptions[@]} " =~ " $filename " ]]; then
```

Comment:
so if I understand this correctly, if a commit hits multiple components we require every component to be added right?
Would it be hard to take the root path from a file and use that automatically?

Reply:
not exactly, I'm checking for a single component
so unless the file is @exception, it will fail if you try for example to update hack and modify ztp in the same commit.
you'd need to break it into 2 atomic commits.

Reply:
There are times where a single logical change/update will span across boundaries. For example when updating the "DU profile" captured in ztp/source-policy-crs (contains symlinks to feature-configs/deploy/....) and ztp/ztp-policy-generator/ranPolicyGenTempExamples. I may need to update the template under the policy generator and have a corresponding change to one of the source policies which is under feature-configs .